### DISCUSSION TITLE
Possibility to add more rules beside "x times in y days"

### DISCUSSION THREAD

**Discussion by fin0** [iSoron/uhabits#203] [Feature Requests]
First, great project here! Thanks to everybody contributing.

I have a small suggestion. The "Hide completed"  option opens the door for one addition that could be useful now: What if you would like to do an activity 2 times per week but NOT on two consecutive days. Like doing a specific sport/training protocol for which you always should have a restday. Would be a nice feature to be able to add this after creating a habit. This would also nicely interact with the nee Hide-feature, because one could hide those habits even if they are not completed for the given time period. (For example: 2 times per week, but not consecutive are the rules. If you did it in tuesday, it should be hidden for wednesday even if you have to do it one more time in the week.)

Hope this is clear. And hope this is no duplicate! Thanks for reading :)

**Comment by adamvoss** [comment]
@fin0 Do you have any suggestions for implementation on this (especially what you envision the UX to be)?  As I was reading your description, it sounded complex like Loop would need a rules engine to handle what users might come up with.

**Comment by ajtga** [comment]
I like the idea. I was thinking about an option that allows the user to select the day of the week that a given habit should appear in the list of habits. For instance, if I want to swim on Tuesdays and Thursdays, the habit "Swim" would not appear in my list on the other days of the week.

**Comment by ocharles** [comment]
My input here would be to follow what `org-habit` (https://orgmode.org/manual/Tracking-your-habits.html) does, which is to let the user say:

> I want to do /x/ at least every /y/ days and at most every /z/ days.

I think a lot of habits I have fall into this pattern. For example, I want to water my plants at least every 10 days, but at most every 5 days. This gives me a 5 day window when I can do the habit, rather having a sudden urgency to do it on a single day. It also removes the situation I just had where I watered my plants on Saturday and was immediately told I should water them again on Sunday (The habit is currently set to 2 times in 10 days).

**Comment by jordise** [comment]
I would like this feature as well. I think an easier solution in terms of programming will be to add the option in periodicity ( daily/ monthly/ X days) letting the user to select the X. This way if I want to do something every 3, days, it is easy to have the % of accomplishment.
I just realized that in yes/no it already exists. Do you know why it does not exist in the measurable option. Would be really nice.
